service-base/src/main/java/org/eclipse/hono/service/tenant/TenantAmqpEndpoint.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.service.tenant;
+
+import static java.net.HttpURLConnection.HTTP_INTERNAL_ERROR;
+
+import java.util.Objects;
+
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.auth.HonoUser;
+import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.service.amqp.RequestResponseEndpoint;
+import org.eclipse.hono.util.MessageHelper;
+import org.eclipse.hono.util.ResourceIdentifier;
+import org.eclipse.hono.util.TenantConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * An {@code AmqpEndpoint} for managing tenant information.
+ * <p>
+ * This endpoint implements Hono's <a href="https://www.eclipse.org/hono/api/TODO/">Tenant API</a>. It receives AMQP 1.0
```

Comment:
I think the URI is pretty clear, isn't it?

Reply:
yes ;-)